### ISSUE TITLE
does it scale?

### ISSUE THREAD

**Issue by sebseb7** [Grant-P-W/GlobalVillagerCuring#3]
When you have a massive amount of players each vilager will have a tag for each player

When you have a massive amount of villagers, each new player joining will update all of them.

Wouldn't it be better to simply get the best reputation value the moment a player opens the trade screen?

**Comment by gwodny** [comment]
Yes, it would be better to get the best reputation value the moment a player opens the trade screen, as long as the trade screen will update the changes to reputation made after it is opened. I have not tested this, so I do not know if it would work or not.

It would also likely be more efficient to store the best reputation values for each villager in a file, so it doesn't have to loop through every player that's joined the server to find the highest reputation each time.

I'm pretty busy right now, but if you want to work on making either or both of these changes, I will merge it. Otherwise, I should get to it in the next couple weeks. And in the meantime, I don't know the specifics of your server, but I think you'd be surprised at how little of an impact on performance this should make even though it is very unoptimized.

This is the first and only MC plugin I've made, and I really just threw it together for a small server of about 25 people 0-7 online at a time and maybe 75–100 villagers, so I wasn't super concerned about scalability. But if other people are actually going to use this, I'd be happy to improve and support it on my free time.

**Comment by sebseb7** [comment]
no worries, take your time.

maybe have a look at this:

https://github.com/kyle-hunady/villager-essentials/blob/main/src/main/java/me/kylehunady/GossipHandler.java

I can't use that plugin because it also does changes to other game mechanics and can't be configured.

My suggestions would be: instead of adding a new gossip target for each player simply update the target of the best major_positive to the the current player in onPlayerInteractAtEntity. That way the nbt data doesn't grow at all over time.
